### ISSUE TITLE
Crash in Support/knowledgeArticles: [SCSArticleQueryResultsController actionCompleted:error:completion:]

### ISSUE THREAD

**Issue by chrisperkins** [forcedotcom/ServiceSDK-iOS#25]
We are receiving crash reports on iPhone & iPad /iOS 11 in SDK 212.0.2/212.0.3. The crashes appear in [SCSArticleQueryResultsController actionCompleted:error:completion: when calling dispatch_group_leave. Stack traces are included below:

```
#3. Crashed: /support/knowledgeArticles
0  libdispatch.dylib              0x183fe3ae4 dispatch_group_leave$VARIANT$mp + 76
1  ServiceKnowledge               0x10310bbd4 -[SCSArticleQueryResultsController actionCompleted:error:completion:] + 456
2  ServiceKnowledge               0x10310b5f8 __76-[SCSArticleQueryResultsController actionToGetFirstPageWithSize:completion:]_block_invoke + 132
3  ServiceCore                    0x102eb716c SFCRandomByteDataWithLength + 111208
4  ServiceCore                    0x102eb874c SFCRandomByteDataWithLength + 116808
5  libdispatch.dylib              0x183fe0b24 _dispatch_call_block_and_release + 24
6  libdispatch.dylib              0x183fe0ae4 _dispatch_client_callout + 16
7  libdispatch.dylib              0x183fe8f18 _dispatch_continuation_pop$VARIANT$mp + 424
8  libdispatch.dylib              0x183fe7850 _dispatch_async_redirect_invoke$VARIANT$mp + 604
9  libdispatch.dylib              0x183fedd30 _dispatch_root_queue_drain + 588
10 libdispatch.dylib              0x183feda80 _dispatch_worker_thread3 + 120
11 libsystem_pthread.dylib        0x184313fac _pthread_wqthread + 1176
12 libsystem_pthread.dylib        0x184313b08 start_wqthread + 4
```

[com.skylifelabs.skyconnecttwu55610_issue_13_crash_1fd77b1f34584cdb84dafc13718a30af_DNE_0_v2-2.txt](https://github.com/forcedotcom/ServiceSDK-iOS/files/2066437/com.skylifelabs.skyconnecttwu55610_issue_13_crash_1fd77b1f34584cdb84dafc13718a30af_DNE_0_v2-2.txt)
[com.skylifelabs.skyconnecttwu55610_issue_13_crash_e447af9fc01947ac9d779073a0c02b54_DNE_0_v2.txt](https://github.com/forcedotcom/ServiceSDK-iOS/files/2066438/com.skylifelabs.skyconnecttwu55610_issue_13_crash_e447af9fc01947ac9d779073a0c02b54_DNE_0_v2.txt)
[com.skylifelabs.skyconnecttwu55610_issue_13_crash_adb09cd428564c8e951124ad056420a5_DNE_0_v2.txt](https://github.com/forcedotcom/ServiceSDK-iOS/files/2066439/com.skylifelabs.skyconnecttwu55610_issue_13_crash_adb09cd428564c8e951124ad056420a5_DNE_0_v2.txt)
[com.skylifelabs.skyconnecttwu55610_issue_13_crash_c4c9d7a8965b4e98ae3604a7c5892651_DNE_0_v2-2.txt](https://github.com/forcedotcom/ServiceSDK-iOS/files/2066440/com.skylifelabs.skyconnecttwu55610_issue_13_crash_c4c9d7a8965b4e98ae3604a7c5892651_DNE_0_v2-2.txt)

**Comment by chrisperkins** [comment]
Also seeing this on iPad now.

[com.skylifelabs.skyconnecttwu55610_issue_13_crash_108a8ffd9ff84d97b952ad5012a985e0_DNE_0_v2.txt](https://github.com/forcedotcom/ServiceSDK-iOS/files/2077987/com.skylifelabs.skyconnecttwu55610_issue_13_crash_108a8ffd9ff84d97b952ad5012a985e0_DNE_0_v2.txt)

**Comment by MikeNachbaur** [comment]
We're looking into this, and should be able to have an answer for you shortly

**Comment by MikeNachbaur** [comment]
@chrisperkins This should be fixed in our latest 214.0.0 release.  If you could verify that, I'd appreciate it.
